trip-ninja-inc/trip_ninja_api_docs
# ENG-672 | initial flight doc update colin waugh

## LocalQA approved by
@sunny-trip-ninja 

## Description
This Ticket : https://app.clickup.com/t/14197839/ENG-672

Start of an update of flight search documentation. 

This began as a catch all update, however a few unexpected updates ended up increasing the changes required. So i decided to split up this ticket from an additional ticket that will take a more granular look at the flight response fields we actually return vs the docs. 
Other Ticket : https://app.clickup.com/t/3dztf6r

- Removed FareStructure and FlexTrip endpoints. 
- Added FareStructure description to initial section.
- Updated search endpoint to search/flights.
- Added route_flexible parameter with old FlexTrip description. 
- Added support for multiple response's to better demonstrate how our response changes with
  - include_itineraries = false/true
  - route_flexible = false/true
- Reworked some segment/itinerary components to create the different response structures.  
- Extracted ErrorDetails into separate component. 
- Added more detail to structure field. 
- Few typo/wording changes.

## How to Test
1. Checkout this branch
2. Run  `npx @redocly/openapi-cli preview-docs tn_api_spec.yml`
3. Go to http://127.0.0.1:8080
4. You should compare with our current documentation, noting the differences described above. 

Current Docs: https://api-documentation.tripninja.io/
